src/sentry/integrations/vsts/webhooks.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import
+from .client import VstsApiClient
+
+from sentry.models import Identity, Integration, sync_group_assignee_inbound
+from sentry.api.base import Endpoint
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+
+import re
+UNSET = object()
+# Pull email from the string: u'lauryn <lauryn@sentry.io>'
+EMAIL_PARSER = re.compile(r'<(.*)>')
+
+
+class WorkItemWebhook(Endpoint):
+    authentication_classes = ()
+    permission_classes = ()
+
+    def get_client(self, identity, oauth_redirect_url):
+        return VstsApiClient(identity, oauth_redirect_url)
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        return super(WorkItemWebhook, self).dispatch(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        data = request.DATA
+        if data['eventType'] == 'workitem.updated':
+            self.handle_updated_workitem(data)
+        return self.respond()
+
+    def handle_updated_workitem(self, data):
+        external_issue_key = data['resource']['workItemId']
+        assigned_to = data['resource']['fields'].get('System.AssignedTo')
+        integration = Integration.objects.get(
+            provider='vsts',
+            external_id=data['resourceContainers']['collection']['id'],
+        )
+        self.handle_assign_to(integration, external_issue_key, assigned_to)
+
+    def handle_assign_to(self, integration, external_issue_key, assigned_to):
+        if not assigned_to:
+            return
+        new_value = assigned_to.get('newValue')
+        if new_value is not None:
+            email = self.parse_email(new_value)
+            assign = True
+        else:
+            email = None
+            assign = False
+        sync_group_assignee_inbound(
+            integration=integration,
+            email=email,
+            external_issue_key=external_issue_key,
+            assign=assign,
+        )
+
+    def parse_email(self, email):
+        return EMAIL_PARSER.search(email).group(1)
+
+    def create_subscription(self, instance, identity_data, oauth_redirect_url, external_id):
+        client = self.get_client(Identity(data=identity_data), oauth_redirect_url)
+        client.delete_all_subscriptions(instance)
```

Comment:
I actually assume this delete function, if it should be used at all, will need to be in the delete flow rather than the creation.... I was just using this for testing. @EvanPurkhiser I know very little about the delete flow or how that works. Can you weigh in?

Reply:
I'm not sure how VSTS' authorization works, but in the scary case they expose all Service Hook subscriptions (not just _ours_) this is probably not a great idea.

Reply:
So here are the two relevant api calls: [Subscription Create](https://docs.microsoft.com/en-us/rest/api/vsts/hooks/subscriptions/list?view=vsts-rest-4.1) and [Subscription Delete](https://docs.microsoft.com/en-us/rest/api/vsts/hooks/subscriptions/delete?view=vsts-rest-4.1). I am a bit confused by the concern over deleting a subscription. For Sentry->Vsts auth we send an oauth access token https://docs.microsoft.com/en-us/vsts/integrate/get-started/authentication/oauth?view=vsts

Reply:
We're not deleting _a_ subscription, we're deleting _all_ subscriptions returned from `https://{account_name}/_apis/hooks/subscriptions` (see https://github.com/getsentry/sentry/pull/8880/files#diff-41b8fbb41a600fb04a5b3844e9d08ca1R260)

That endpoint returns all Service Hook subscriptions for the account – ones created manually via the UI, ones created by other services (I'm guessing), ones created by Sentry, etc, as far as I can tell. We're potentially deleting a bunch of unrelated subscriptions by doing it this way.

We should either store the Subscription's ID somewhere or filter through the response from that endpoint and find the one we care about (via `consumerInputs.url`).

Reply:
ooohhh I see. Thanks for the catch! That's not great.